### PR TITLE
Mixin 0.8 preview adaption

### PR DESCRIPTION
**Pull request by liach**
Updated the option return method and removed the mixin service (seems only used for runtime, not ap)
Signed-off-by: liach <liach@users.noreply.github.com>

### CODE REVIEW

**Comment by modmuss50** [comment]
In loom we will need to tweak how this is pulled in to ensure the correct one is used for the correct mixin version

**Comment by modmuss50** [comment]
Looking at it, it wouldnt be too hard to support old and new, that may be a better idea.

File: src/main/java/net/fabricmc/loom/mixin/ObfuscationServiceFabric.java [hunk 1]
```diff
@@ -69,6 +70,11 @@ private void addSupportedOptions(ImmutableSet.Builder<String> builder, String fr
 	}
 
 	@Override
```
Comment (modmuss50) [inline src/main/java/net/fabricmc/loom/mixin/ObfuscationServiceFabric.java:72]: Not sure if this needs to be gone? for 0.6 support? Does java check @Override at runtime? im not sure.

Comment (liach) [reply src/main/java/net/fabricmc/loom/mixin/ObfuscationServiceFabric.java:72]: image

**Comment by modmuss50** [comment]
Humm, we may need a version of this PR for a branch before the v2 tiny changes for older yarn versions.

**Comment by liach** [comment]
This compiler extensions should be compatible with old yarn afaik as the mapping parser's tree api can load from v1 mappings.

**Comment by modmuss50** [comment]
Were the changes to parser backwards compatible? as it will need to use the new parser?

**Comment by liach** [comment]
The new parser 0.2 did not remove anything from the parser 0.1.

**Comment by modmuss50** [comment]
Ah, thats fine then. Nice work.

**Pull request merged**